ex2_hal/iris/equipment_handler/source/iris_spi.c

```diff
@@ -138,9 +122,10 @@ void iris_spi_get(uint16_t *rx_data, uint16_t data_length) {
  *   Number of ticks for non FreeRTOS delay
  *
  **/
-void iris_spi_delay(uint16_t ticks) {
+void iris_spi_delay(uint32_t ticks) {
```

Comment:
Why are you using a loop delay? This is unreliable in an operating system environment and may get optimized out completely

Reply:
Due to the nature of the spi interface between OBC and Iris, we sometime need delays less than 1ms. For instance, the time delay between the last clock edge and NSS going back to Vdd does not need to be too long, it will a matter of microseconds. Basically, it gives me a finer spi control

Reply:
You should check if this behaves as expected with -O3 before we merge